xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/macro/descriptor/SyntaxDescriptor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.macro.descriptor;
+
+import java.util.Map;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.rendering.syntax.Syntax;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Represents the syntax of some content of the macro.
+ *
+ * @version $Id$
+ * @since 10.9-rc-1
+ */
+@Unstable
+@Role
+public interface SyntaxDescriptor
+{
+    /**
+     * @param macroParameters the parameters of the macro to be used.
+     * @return the syntax of the macro computed from information given in its parameters.
+     */
+    Syntax getSyntax(Map<String, Object> macroParameters);
```

Comment:
How is the WYSIWYG supposed to use that ?

Reply:
AFAI understood @vmassol proposal the idea is to retrieve the component with the hint (see https://github.com/xwiki/xwiki-rendering/pull/143/files/11802d7856f36c09831c07a2231e396c082d29cf#diff-01939d2b6482e4ca74d1e171539afaad) and I assume the WYSIWYG already have the map of parameters with their value: then the component is able to determine the syntax and to return it. 

I put this proposal on purpose in the PR to be able to discuss it, but I also can remove it and put this in another PR to avoid mixing things: as @vmassol stated it could be a step 2.

Reply:
> but I also can remove it and put this in another PR to avoid mixing things

Would be better yes.

Reply:
OK, I'll do that then